syncd/FlexCounter.cpp

```diff
@@ -2023,6 +2320,24 @@ void FlexCounter::updateSupportedBufferPoolCounters(
     }
 }
 
+void FlexCounter::updateSupportedBufferPoolCounters(
+        _In_ sai_object_id_t bufferPoolId,
+        _In_ const std::vector<sai_buffer_pool_stat_t> &counterIds,
+        _In_ sai_stats_mode_t statsMode)
+{
+    SWSS_LOG_ENTER();
+
+    m_supportedBufferPoolCounters.clear();
+
+    /* Query SAI supported buffer pool counters */
+    sai_status_t status = querySupportedBufferPoolCounters(bufferPoolId, statsMode);
+    if (status != SAI_STATUS_SUCCESS)
+    {
+        /* Fallback to legacy approach */
```

Comment:
why do we have 2 approaches?

Reply:
The legacy approach is the one we used before.
The new one is not implemented for all vendors, which means a vendor can choose to return NOT_IMPLEMENTED and keep the old way.